src/arnParser.js

```diff
@@ -1,12 +1,13 @@
-const ARN_VALIDATOR = /^arn:aws:kinesis:([a-z\-0-9]+):(\d+):stream\/(.+)$/;
+const ARN_VALIDATOR =
+  /^arn:aws:kinesis:([a-z\-0-9]+):(?:(?:http(?:s):\/\/.)[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b(?:[-a-zA-Z0-9@:%_\+.~#?&//=]*)|(?:\d+)):stream\/(.+)$/;
```

Comment:
I don't think this needs to be extended here. A Kinesis stream ARN does not contain http URLs like this, even if it's created in localstack.

The application should be configurable to send requests to a custom endpoint when testing instead.

Reply:
Agreed! Thank you for the helpful redirection. Closing this PR 😊